src/artemis/external_interaction/callbacks/fgs/zocalo_callback.py

```diff
@@ -88,6 +88,34 @@ def wait_for_results(self, fallback_xyz: Point3D) -> Point3D:
                 datacollection_group_id
             )
 
+            # Sort from strongest to weakest in case of multiple crystals
+            raw_results = sorted(
+                raw_results, key=lambda d: d["total_count"], reverse=True
+            )
+            LOGGER.info(f"Zocalo: found {len(raw_results)} crystals.")
+            multi_crystal_msg = (
+                f"Found multiple crystals: {len(raw_results)}."
+                if len(raw_results) > 1
+                else ""
+            )
+
+            bboxes = []
+            for n, res in enumerate(raw_results):
+                bboxes.append(
+                    list(
+                        map(
+                            operator.sub, res["bounding_box"][1], res["bounding_box"][0]
+                        )
+                    )
+                )
+                multi_crystal_msg += (
+                    f"Crystal {n+1}: "
+                    f"Strength {res['total_count']} "
+                    f"Position (x,y,z) {res['centre_of_mass']} "
+                    f"Size (x,y,z) {bboxes[n]} "
+                )
```

Comment:
Should: This is potentially going to be hard to read, can we put in a carriage return between each crystal?

Reply:
Done, good point